File: Leetcode/ARRAY/JavaScript/Medium Problems/P_11.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let right = height.length - 1,
    left = 0,
    max_Area = 0;
  while (left < right) {
    let current_area = Math.min(height[left], height[right]) * (right - left);
    if (max_Area < current_area) max_Area = current_area;
    if (height[left] < height[right]) left++;
    else right--;
  }
  return max_Area;
};

let height = [1, 8, 6, 2, 5, 4, 8, 3, 7];
console.log(maxArea(height));

File: Leetcode (DSA)/ARRAY/JAVA/Easy Problems/P_11.java
public class P_11 {
    public static int maxArea(int[] height) {
        int maxWater = 0;

        int right = height.length - 1;
        int left = 0;

        while (left < right) {

            int minHeight = 0;

            if (height[left] < height[right]) {
                minHeight = height[left];
            } else {
                minHeight = height[right];
            }

            int area = minHeight * (right - left);

            if (area > maxWater) {
                maxWater = area;
            }
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxWater;
    }

    public static void main(String[] args) {
        int[] height = { 1, 8, 6, 2, 5, 4, 8, 3, 7 };
        System.out.println(maxArea(height));
    }
}
